src/backends/perf_pt/collect.c

```diff
@@ -493,7 +493,7 @@ tracer_thread(void *arg)
 
 clean:
     if (!sem_posted) {
-        assert(!ret);
+        xassert(!ret, "Expected failure return code");
```

Comment:
That's checking that we only unlock the semaphore if we didn't fail. So that's an internal consistency check.

Reply:
Fair point: "internal" and "external" aren't good words. Yes, this is internal to our address space; but it's an external check in the sense that it's outside our source code. Also, it's expected that sometimes it might fail, and that we might (one day) be able to recover from it.

Reply:
Yup, and that's fine. When that day comes `xassert` will give me a good error message and file/line and we can make the change.

I think you are being too philosophical :)